bin/evaluate.sh

```diff
@@ -2,6 +2,13 @@
 
 set -o errexit
 
+backtrace="--backtrace"
+if [ "${*#*$backtrace}" != "$*" ]; then
+    export RUST_BACKTRACE=1
+    set -- ${*%$backtrace*}${*#*$backtrace}
```

Comment:
**tl;dr:** I'll pass `--backtrace` as first arg, and remove it with `shift`, and also simplify the `if` test (although I borrowed the code for `if` from the below `--test` and `--color=always` code)  which results in simpler code!

-----------
Don't read this, but I'll leave it here anyway :)

I used a simpler syntax in this commit: https://github.com/rust-lang/rust-playpen/pull/191/commits/a147805ca4e75f785293bf4068025dd2c773f0e3
But that had the limitation that `--backtrace` must always be the first passed arg! else it would break things badly(eg. delete another arg!) So, if someone would modify web.py not knowing `--backtrace` must be first arg, then things would break so badly...

The ugly bash syntax(actually, it's running under the `dash` shell - so there are no arrays for example) doesn't have the above limitation and it's making sure that when `--backtrace` is specified wherever in the args list it will be safely removed, UNLESS it's specified more than once, then that will break. But I don't see why it would ever be specified more than once, unless a mistake in web.py  but still not as easy of a mistake to make than in the above case with `--backtrace` having to me first arg.

`set -- ${*%$backtrace*}${*#*$backtrace}`
This resets the `$*` args (eg. it rewrites them) to have `--backtrace` removed by concatenating the two halves of the current `$*`, eg.
if `$*` is initially `somearg --test --backtrace --color=always "arg with space" moreargs here`
and `$#` is thus `7`
The `set ` above will concat `somearg --test ` with `--color=always "arg with space" moreargs here` and set the result as being the new `$*` (the args passed to `evaluate.sh` are thus modified/reset to this new value)
and still ensure the number of args is *the same minus 1*, even if args with spaces exist!
So it becomes: `set -- somearg --test  --color=always "arg with space" moreargs here`
thus `$*` is now `somearg --test  --color=always "arg with space" moreargs here`
`$#` is `6`

Actually I forgot to test for args with spaces and it's clearly not working for that case(`$#` is `9`!)! eg. args with spaces are split into more than one arg. eg. the one argument "arg with space" becomes 3 args: "arg" "with" "space"
But I thought I did test it...

It doesn't seem to be a way to also remove `--backtrace` AND keep the args with spaces intact, and I am unsure if it's necessary to keep them intact that way...  they are only needed for this line, currently: `TERM=xterm rustc - -o ./out "$@"` In other words, if `rustc` currently(or in the future!) accepts any args with spaces(not counting the filename which is stdin here), then it's a MUST that I have to keep the args intact!  I'll investigate more...

Worst case, we could revert to that that commit, it's the simpler syntax(as you'd want) AND it keeps the args intact! Hmm... I could even just check for `$1`, expecting `--backtrace` to be the first, and only then remove it, and if it's not first, then it will cause rustc to fail because it will get `--backtrace` passed to it ... yes... this seems like a simpler way, an interestingly simple compromise between what I want and what you want:)